library/std/src/os/windows/io/handle.rs

```diff
@@ -91,7 +92,7 @@ pub struct OwnedHandle {
 #[repr(transparent)]
 #[stable(feature = "io_safety", since = "1.63.0")]
 #[derive(Debug)]
-pub struct HandleOrNull(OwnedHandle);
+pub struct HandleOrNull(ManuallyDrop<OwnedHandle>);
```

Comment:
I'd simply use `RawHandle` here, so that we don't have to do the `ManuallyDrop::take` dance below.

Reply:
Done. Since it's not possible to move out the inner field with a custom `Drop` implementation, I was looking for a decent way to implement `TryFrom`. However, I can just forget the struct altogether.